dj/construction/build.py

```diff
@@ -339,47 +343,144 @@ def _build_select_ast(
+
     if filters:
         filter_asts = (  # pylint: disable=consider-using-ternary
             query.select.where and [query.select.where] or []
         )
 
         for filter_ in filters:
+            # use parse to get the asts from the strings we got
             temp_select = parse(f"select * where {filter_}").select
             filter_asts.append(
-                # use parse to get the asts from the strings we got
                 temp_select.where,  # type:ignore
             )
+            for col in temp_select.find_all(ast.Column):
+                projection_addition[col.identifier(False)] = col
+                col.namespace_table()
+                if col.table:#pragma: no cover
+                    dimension_tables.add(cast(ast.Table, col.table).identifier(False))
         query.select.where = ast.BinaryOp.And(*filter_asts)
 
-    if dimensions:
-        for agg in dimensions:
-            temp_select = parse(
-                f"select * group by {agg}",
-            ).select
-            query.select.group_by += temp_select.group_by  # type:ignore
-            projection_addition += list(temp_select.find_all(ast.Column))
-    query.select.projection += list(projection_addition)
+    if not query.organization:
+        query.organization = ast.Organization([])
+
+    # for order bys, we must be using dimensions
+    # that are already used in dimensions, filters
+    # so for the sake of building rn, we will dupe cols
+    # if used already, otherwise, the orderby col will
+    # be put in the projection as with the dimensions, filters
+    # columns
+    orderby_tables = set()
+    if orderby:
+        for order in orderby:
+            temp_query = parse(
+                f"select * order by {order}",
+            )
 
-    # Cannot select for columns that aren't in GROUP BY and aren't aggregations
-    if query.select.group_by:
-        query.select.projection = [
-            col
-            for col in query.select.projection
-            if col.is_aggregation()  # type: ignore
-            or col.name.name in {gc.name.name for gc in query.select.group_by}  # type: ignore
-        ]
+            for col in temp_query.find_all(ast.Column):
+                ident = col.identifier(False)
+                col.namespace_table()
+                if ident in projection_addition:
+                    col.swap(projection_addition[ident])
+                else:
+                    projection_addition[ident] = col
+                if col.table:#pragma: no cover
+                    orderby_tables.add(cast(ast.Table, col.table).identifier(False))
+            
+            query.organization.order += temp_query.organization.order  # type:ignore
+
+    if diff := orderby_tables - dimension_tables:
+        raise DJInvalidInputException(
+            "Order by columns can only be from "
+            "dimensions used in filters or dimensions arguments"
```

Comment:
Should "dimensions arguments" just be called "group bys" instead? (I'm assuming that most of users would understand that better since it's largely a SQL-based system, whereas the distinction between `dimension arguments` and `dimensions` is not super clear)

Reply:
@shangyian are you suggesting to change all the api arguments from `dimensions` to `group_bys` or just the terms in that error